Sources/MapboxNavigation/Route.swift

```diff
@@ -43,6 +43,68 @@ extension Route {
         }
     }
     
+    func congestionFeatures(legIndex: Int? = nil,
+                            isAlternativeRoute: Bool = false,
+                            roadClassesWithOverriddenCongestionLevels: Set<MapboxStreetsRoadClass>? = nil) -> [Feature] {
+        guard let coordinates = shape?.coordinates, let shape = shape else { return [] }
+        var features: [Feature] = []
+        
+        for (index, leg) in legs.enumerated() {
+            let legFeatures: [Feature]
+            
+            if let congestionLevels = leg.segmentCongestionLevels, congestionLevels.count < coordinates.count {
+                // The last coordinate of the preceding step, is shared with the first coordinate of the next step, we don't need both.
+                let legCoordinates: [CLLocationCoordinate2D] = leg.steps.enumerated().reduce([]) { allCoordinates, current in
+                    let index = current.offset
+                    let step = current.element
+                    let stepCoordinates = step.shape!.coordinates
+                    
+                    return index == 0 ? stepCoordinates : allCoordinates + stepCoordinates.suffix(from: 1)
+                }
+                
+                let mergedCongestionSegments = legCoordinates.combined(congestionLevels,
+                                                                       streetsRoadClasses: leg.streetsRoadClasses,
+                                                                       roadClassesWithOverriddenCongestionLevels: roadClassesWithOverriddenCongestionLevels)
+                
+                legFeatures = mergedCongestionSegments.map { (congestionSegment: CongestionSegment) -> Feature in
+                    var feature = Feature(LineString(congestionSegment.0))
+                    feature.properties = [
+                        CongestionAttribute: String(describing: congestionSegment.1),
+                        "isAlternativeRoute": isAlternativeRoute,
+                        CurrentLegAttribute: (legIndex != nil) ? index == legIndex : index == 0
+                    ]
+                    
+                    return feature
+                }
+            } else {
+                var feature = Feature(LineString(shape.coordinates))
+                feature.properties = [
+                    "isAlternativeRoute": isAlternativeRoute,
+                    CurrentLegAttribute: (legIndex != nil) ? index == legIndex : index == 0
+                ]
+                legFeatures = [feature]
+            }
+            
+            features.append(contentsOf: legFeatures)
+        }
+        
+        return features
+    }
+    
+    func identifier(_ routeLineType: RouteLineType) -> String {
```

Comment:
This method should become a method of RouteLineType itself that takes a base identifier as an argument.

Reply:
I don't think it'll allow us to make implementation clear and understandable. In my opinion `RouteLineType` should not have identifier, but `Route` object should be responsible for storing its identifier based on specific conditions (passed via `RouteLineType`). If we move identification logic to `RouteLineType` and store base identifier separately it'll not be really clear, in current case both logic and data reside in one entity.